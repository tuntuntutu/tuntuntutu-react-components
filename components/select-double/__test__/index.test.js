/* eslint-disable */
import React from 'react';
import Double from '../index';
import sinon from 'sinon'
import Enzyme, {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer';

const prefix = 'http://devcenter.caocaokeji.cn'
const provinces = [
  {
    "regionCode": "110000",
    "telCode": "010",
    "fullName": "北京市",
    "name": "北京市"
  },
  {
    "regionCode": "120000",
    "telCode": "022",
    "fullName": "天津市",
    "name": "天津市"
  }, {
    "regionCode": "130000",
    "telCode": null,
    "fullName": "河北省",
    "name": "河北省"
  },];

const cities = [
  {
    "regionCode": "140100",
    "telCode": "0351",
    "fullName": "山西省太原市",
    "name": "太原市"
  },
  {
    "regionCode": "140200",
    "telCode": "0352",
    "fullName": "山西省大同市",
    "name": "大同市"
  }, {
    "regionCode": "140300",
    "telCode": "0353",
    "fullName": "山西省阳泉市",
    "name": "阳泉市"
  },];
const config = {
  width: 200,
  className: '',
  firstConfig: {
    url: {
      method: 'get',
      url: `${prefix}/car-insurance/regionCode/queryAllProvinces`,
    },
    optionMap: {key: 'regionCode', value: 'name'},
  },
  secondConfig: {
    url: {
      method: 'get',
      url: `${prefix}/car-insurance/regionCode/queryCities`,
    },
    queryKey: 'provinceCode',
    optionMap: {key: 'telCode', value: 'name'},
  },
  defaultValue: {},
  onChange: () => {
  },
  value: {},
  format: {
    first: 'provinceCode',
    second: 'cityCode',
  },
  connect: () => {
  },
}

describe('SelectDouble组件', () => {
  const mountDouble = () => mount(<Double
    {...config}
  />);

  it('render SelectDouble correctly', () => {
    const double = <Double
      {...config}
    />;

    expect(renderer.create(double).toJSON()).toMatchSnapshot();

    const wrapper = mountDouble();

    expect(wrapper.find('Trigger').length).toBe(2);
    expect(wrapper.find('SelectTrigger').length).toBe(2);
    sinon.spy(wrapper.instance(), 'getSecondList');
    expect(wrapper.instance().getSecondList.calledOnce).toBeTruthy;

    const wrapperLink = shallow((double));

    expect(wrapperLink.find('div').children().length).toBe(2);
  });

  it('func is triggered correctly', () => {
    const wrapper = mountDouble();
    sinon.spy(wrapper.instance(), 'getFirstList');
    expect(wrapper.instance().getFirstList.calledOnce).toBeTruthy;
  })
});