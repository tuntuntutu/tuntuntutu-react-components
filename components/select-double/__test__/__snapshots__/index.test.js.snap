// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDouble组件 render SelectDouble correctly 1`] = `
<div
  className=""
>
  <div
    className="ant-select ant-select-enabled ant-select-allow-clear"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "width": 200,
      }
    }
  >
    <div
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup="true"
      className="ant-select-selection
            ant-select-selection--single"
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <div
        className="ant-select-selection__rendered"
      >
        <div
          className="ant-select-selection__placeholder"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "display": "block",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          请选择
        </div>
        <div
          className="ant-select-search ant-select-search--inline"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="ant-select-search__field__wrap"
          >
            <input
              autoComplete="off"
              className="ant-select-search__field"
              onChange={[Function]}
              onKeyDown={[Function]}
              value=""
            />
            <span
              className="ant-select-search__field__mirror"
            >
              
               
            </span>
          </div>
        </div>
      </div>
      <span
        className="ant-select-arrow"
        onClick={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <b />
      </span>
    </div>
  </div>
  <div
    className="ant-select ant-select-enabled ant-select-allow-clear"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "width": 200,
      }
    }
  >
    <div
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup="true"
      className="ant-select-selection
            ant-select-selection--single"
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <div
        className="ant-select-selection__rendered"
      >
        <div
          className="ant-select-selection__placeholder"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "display": "block",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          请选择
        </div>
        <div
          className="ant-select-search ant-select-search--inline"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="ant-select-search__field__wrap"
          >
            <input
              autoComplete="off"
              className="ant-select-search__field"
              onChange={[Function]}
              onKeyDown={[Function]}
              value=""
            />
            <span
              className="ant-select-search__field__mirror"
            >
              
               
            </span>
          </div>
        </div>
      </div>
      <span
        className="ant-select-arrow"
        onClick={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <b />
      </span>
    </div>
  </div>
</div>
`;
